src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerCache.cs

```diff
@@ -43,14 +57,13 @@ private InnerCache CurrentCache
             }
         }
 
-        public ControllerActionInvokerState GetState(ControllerContext controllerContext)
+        public (ControllerActionInvokerState State, IFilterMetadata[] Filters) GetState(ControllerContext controllerContext)
```

Comment:
We need to return two pieces of data (the cache entry and the filters. Any suggestions on how you'd want to surface this?